bangc-ops/mlu_op.h

```diff
@@ -2558,8 +2558,86 @@ mluOpStatus_t MLUOP_WIN_API mluOpYoloBox(
     const float iou_aware_factor, const mluOpTensorDescriptor_t boxes_desc,
     void *boxes, const mluOpTensorDescriptor_t scores_desc, void *scores);
 
+/*!
+ * @brief Taking the point's index in the \b new_xyz set as the center of the sphere,
+ * and using \b min_radius and \b max_radius as the radius, returns the \b idx of
+ * the first \n nsample points in the \b xyz set in the spherical domain.
+ *
+ * @param[in] handle
+ * Handle to an MLUOP context that is used to manage MLU devices and
+ * queues in the sqrt backward operation. For detailed information, see
+ * ::mluOpHandle_t.
+ * @param[in] new_xyz_desc
+ * The descriptor of the new_xyz tensors, which indicates the center of the ball.
+ * For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[in] new_xyz
+ * Pointer to the MLU memory that stores the new_xyz tensor.
+ * @param[in] xyz_desc
+ * The descriptor of the xyz tensors, which means cloud points. For detailed information,
+ * see ::mluOpTensorDescriptor_t.
+ * @param[in] xyz
+ * Pointer to the MLU memory that stores the xyz tensor.
+ * @param[in] min_radius
+ * An float value which is the smaller radius.
+ * @param[in] max_radius
+ * An float value which is the bigger radius.
+ * @param[in] nsample
+ * An integer value which is the selected points index.
+ * @param[in] idx_desc
+ * The descriptor of the idx tensors, which contains output indexs. For detailed information,
+ * see ::mluOpTensorDescriptor_t.
+ * @param[in] idx
+ * Pointer to the MLU memory that stores the xyz tensor.
+ *
+ * @par Return
+ * - ::MLUOP_STATUS_SUCCESS, ::MLUOP_STATUS_BAD_PARAM
+ *
+ * @par Data Type
+ * - The supported data types of new_xyz tensor \b new_xyz, xyz tensor \b xyz and
+ *   idx tensor \b idx are as fllows:
+ *   - new_xyz tensor: float or half.
+ *   - xyz tensor: float or half.
+ *   - idx tensor: int.
+ *
+ * @par Scale Limitation
+ * - The new_xyz tensor, xyz tensor and idx tensor must be 3-D.
+ * - The first dimension of the new_xyz tensor, xyz tensor and the idx tensor must be the same.
+ * - The second dimension of the new_xyz tensor and the idx tensor must be the same.
+ * - The third dimension of the new_xyz tensor and the xyz tensor must be the same and equal to 3.
+ * - The third dimension of idx tensor must be equal to nsample.
+ * - The \b min_radius should be larger than 0.
+ * - The \b max_radius should be larger than 0.
+ * - The \b nsample should be larger than 0.
+
+ * @note
+ * - Taking the point in new_xyz as the center of the sphere, There are no points in xyz within a
+ *   sphere with min_radius and max_radius as diameters.At this time, the value of the corresponding
+ *   position in idx is the value when it is passed into the kernel. In the cuda, Generally, before
+ *   passing idx into the kernel, using cnrtMemset brushe all the values in idx to 0.
```

Comment:
Take the point in new_xyz as the center of the sphere, there may be no points in xyz within the
sphere with min_radius and max_radius as diameters. At this time, the value of the corresponding
position in idx is the value when it is passed into the kernel. Generally, before
passing idx into the kernel, use cnrtMemset to brush all the values in idx to 0.